src/CollateralizedVault.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.13;
+
+import "yield-utils/contracts/math/WDiv.sol";
+import "yield-utils/contracts/math/WMul.sol";
+import "yield-utils/contracts/token/IERC20.sol";
+import "yield-utils/contracts/token/TransferHelper.sol";
+import "chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "yield-utils/contracts/access/Ownable.sol";
+
+/// https://github.com/yieldprotocol/mentorship2022/issues/5
+
+contract CollateralizedVault is Ownable {
+    
+    using TransferHelper for IERC20;
+
+    /// mainnet 0x6B175474E89094C44Da98b954EedeAC495271d0F
+    /// rinkeby 0x0165b733e860b1674541BB7409f8a4743A564157
+    address DAI;
+
+    /// mainnet 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
+    /// rinkeby 0xDf032Bc4B9dC2782Bb09352007D4C57B75160B15
+    address WETH;
+
+    /// mainnet 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48
+    /// rinkeby 0xab5400b26149A3fF5918EFCdeB2C37903042E9ee
+    address USDC;
+
+    /// mainnet 0x773616E4d11A78F511299002da57A0a94577F1f4
+    /// rinkeby 0x74825DbC8BF76CC4e9494d0ecB210f676Efa001D
+    /// DAI/ETH Price Oracle
+    
+    /// mainnet 0x986b5E1e1755e3C2440e960477f25201B0a8bbD4
+    /// rinkeby 0xdCA36F27cbC4E38aE16C4E9f99D39b42337F6dcf
+    /// USDC/ETH Price Oracle
+
+    /// @dev mapping for storing Chainlink Oracle Interfaces
+    /// key is address of a token for a token/ETH feed
+    mapping(address => AggregatorV3Interface) public EthFeeds;
+    
+
+    /// had been stored in a struct but changed after Alberto suggested nested mappings in Slack
+    /// @dev nested mapping for tracking user debts
+    mapping(address => mapping(address => uint256)) public Debts;
+
+    /// @dev nested mapping for tracking user deposits
+    mapping(address => mapping(address => uint256)) public Deposits;
+
+    
+
+    event Deposit(address indexed user, uint256 amount);
+    event Borrow(address indexed user, uint256 amount);
+    event Repay(address indexed user, uint256 repaid, uint256 remaining);
+    event Withdraw(address indexed user, uint256 amount);
+
+    constructor (address dai_, address weth_, address usdc_, AggregatorV3Interface dETH_, AggregatorV3Interface uETH_) {
+        DAI = dai_;
+        WETH = weth_;
+        USDC = usdc_;
+        EthFeeds[DAI] = dETH_;
+        EthFeeds[USDC] = uETH_;
+    }
+
+    /// @return price the price of 1 DAI in WETH
+    function _daiPrice() internal view returns(uint256) {
```

Comment:
You could have a `_price(address token)` function, instead of `_daiPrice()` and `usdcPrice()`.

That could go on and you could have an `_available(address token, address user)` function.

Then you could have a `convert(address fromToken, address toToken, uint256 amount)` function, that works DAI, USDC and WETH in any combination.

And finally, you can have `borrow(address token, uint256 amount)` and `repay(address token, uint256 amount).